language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/executors/PullPackageExecutor.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ballerinalang.langserver.command.executors;
+
+import com.google.gson.internal.LinkedTreeMap;
+import org.apache.commons.io.IOUtils;
+import org.ballerinalang.annotation.JavaSPIService;
+import org.ballerinalang.langserver.command.ExecuteCommandKeys;
+import org.ballerinalang.langserver.command.LSCommandExecutor;
+import org.ballerinalang.langserver.command.LSCommandExecutorException;
+import org.ballerinalang.langserver.common.constants.CommandConstants;
+import org.ballerinalang.langserver.common.utils.CommonUtil;
+import org.ballerinalang.langserver.compiler.LSCompiler;
+import org.ballerinalang.langserver.compiler.LSContext;
+import org.ballerinalang.langserver.diagnostic.DiagnosticsHelper;
+import org.eclipse.lsp4j.MessageType;
+import org.eclipse.lsp4j.services.LanguageClient;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import static org.ballerinalang.langserver.command.CommandUtil.clearDiagnostics;
+import static org.ballerinalang.langserver.command.CommandUtil.notifyClient;
+
+/**
+ * Command executor for pulling a package from central.
+ *
+ * @since 0.983.0
+ */
+@JavaSPIService("org.ballerinalang.langserver.command.LSCommandExecutor")
+public class PullPackageExecutor implements LSCommandExecutor {
+
+    // A newCachedThreadPool with a limited max-threads
+    private static ExecutorService executor = new ThreadPoolExecutor(0, Runtime.getRuntime().availableProcessors(), 60L,
+            TimeUnit.SECONDS, new SynchronousQueue<>());
+
+    private static final String COMMAND = "PULL_MODULE";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Object execute(LSContext context) throws LSCommandExecutorException {
+        executor.submit(() -> {
+            // Derive package name and document uri
```

Comment:
Since we moving away from 'package', shall we update this comment to be 'Derive module name and document uri'?

Reply:
Fixed